packages/core/lib/debug/interpreter.js

```diff
@@ -132,42 +152,76 @@ class DebugInterpreter {
 
       if (matchingSources.length === 0) {
         this.printer.print(`No source file found matching ${sourceArg}.`);
-        return;
+        return [];
       } else if (matchingSources.length > 1) {
-        this.printer.print(
-          `Multiple source files found matching ${sourceArg}.  Which did you mean?`
-        );
-        matchingSources.forEach(source =>
-          this.printer.print(source.sourcePath)
-        );
-        this.printer.print("");
-        return;
+        //normally if there's multiple matching sources, we want to return no
+        //breakpoint and print a disambiguation prompt.
+        //however, if one of them has a source path that is a substring of all
+        //the others...
+        if (
+          matchingSources.some(
+            shortSource => matchingSources.every(
+              source =>
+                typeof source.sourcePath !== "string" //just ignore these I guess?
+                || source.sourcePath.includes(shortSource.sourcePath)
+            )
+          )
+        ) {
+          //exceptional case
+          this.printer.print(
+            `WARNING: Multiple source files matched, but one is a proper substring of all others, so disambiguation is not possible.  Acting on all of them.`
```

Comment:
I think this can be simplified:
```suggestion
            `WARNING: Multiple source files matched; could not disambiguate.  Setting breakpoints for:`
```

Reply:
Well, we can't go with this exactly.  It can't be "setting breakpoints for", because this also applies to *removing* breakpoints.  We can have it switch as appropriate (I didn't bother as you can see, but it wouldn't be hard), but it can't just be "setting".

Also, I don't like "setting breakpoints for:" because the normal messages will still print for each.  If it were "setting breakpoints for:", you would expect to just see the *location* printed for each, not the full message.

Also, I think we do need a "but" in there.  "Multiple source files matched; could not disambiguate" sounds like we don't perform disambiguation.  We need to say "Multiple source files matced, but could not disambiguate" to make it clear that we would disambiguate if we could.

...although, really, *we* don't disambiguate, we ask the *user* to disambiguate.  Hence why I said "disambiguation is not possible", not "could not disambiguate".  IDK, maybe that's getting too nitpicky.

Reply:
Does the user need to know the internals? Can we inform the user we encountered ambiguity because of multiple source name collision and set expectations that way?

Reply:
Thanks for the clarification, it makes sense. How about rewording to present course of action followed by reasoning? 

> WARNING: acting on all sources because disambiguation is not possible due to one filename is a proper substring of all others.

Reply:
So, rewording a little, if we take @cds-amal's approach, but also shorten things a bit as @gnidan wanted, I might write something like...

> WARNING: Acting on all matching sources because disambiguation between them is not possible.

How does that sound? :)

Reply:
I'm ok with that!

Reply:
OK, I've made this change.